cyclonedx/validation/__init__.py

```diff
@@ -27,13 +28,75 @@
     from .xml import XmlValidator
 
 
+def squeeze(text: str, size: int, replacement: str = ' ... ') -> str:
+    """Replaces the middle of ``text`` with ``replacement``.
+
+    :param size: the length of the output, -1 to make no squeezing.
+    :return: potentially shorter text
+    :retval: ``text`` if ``size`` is -1 (for easy pass-through)
+    :retval: ``text`` if it is shorter than ``size``
+    :retval: ``text`` with the middle of it replaced with ``replacement``,
+             if ``text`` is longer, than ``size``
+
+    Raises error if ``replacement`` is longer than ``size``, and replacement
+    would happen.
+    """
+    if size == -1:
+        return text
+
+    if size < len(replacement):
+        raise ValueError(f'squeeze: {size = } < {len(replacement) = }')
+
+    if len(text) <= size:
+        return text
+
+    left_size = (size - len(replacement)) // 2
+    right_size = size - len(replacement) - left_size
+    right_offset = len(text) - right_size
+
+    return f'{text[:left_size]}{replacement}{text[right_offset:]}'
+
+
 class ValidationError:
     """Validation failed with this specific error.
 
-    Use :attr:`~data` to access the content.
+    You can use :attr:`~data` to access the raw error object, but prefer
+    other properties and functions, if possible.
     """
 
     data: Any
+    """Raw error data from one of the validation libraries."""
+
+    @property
+    def message(self) -> str:
+        """The error message."""
+        return str(getattr(self.data, 'message', self))
+
+    @property
+    def path(self) -> str:
+        """Path to the location of the problem in the document.
+
+        An XPath/JSONPath string.
+        """
+        # only subclasses know how to extract this info
+        return str(getattr(self.data, 'path', ''))
```

Comment:
your implementation clearly shows, that this has no general value, as it gets overridden in every second implementation.

why have it in the general implementation, if it only holds for SOME contexts?

Reply:
It shows the expected implementation.
`jsonschema` has it as `json_path` instead of `path`.
`lxml` could also define it as `xpath`, but is not.

I am fine splitting it into 2 optional properties (`str | None`) if that looks more acceptable:
- `json_path`
- `xpath`